core/src/main/scala/cats/Foldable.scala

```diff
@@ -247,6 +247,14 @@ import Foldable.sentinel
    */
   def combineAll[A: Monoid](fa: F[A]): A = fold(fa)
 
+  def combineAllOption[A](fa: F[A])(implicit ev: Semigroup[A]): Option[A] =
+    if (isEmpty(fa)) None else ev.combineAllOption(toIterator(fa))
+
+  def toIterator[A, B](fa: F[A]): Iterator[A] =
```

Comment:
Could you override the implementations in the `List`, etc. instances to use the standard methods? We'll also need to change the implementation here now that `Stream` is deprecated in 2.13.